sdk/azidentity/azure_developer_cli_credential.go

```diff
@@ -0,0 +1,170 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+const (
+	credNameAzureDeveloperCLI = "AzureDeveloperCLICredential"
+	timeoutAzdRequest         = 10 * time.Second
+)
+
+// used by tests to fake invoking the CLI
+type azureDeveloperCLITokenProvider func(ctx context.Context, scope []string, tenantID string) ([]byte, error)
+
+// AzureDeveloperCLICredentialOptions contains optional parameters for AzureDeveloperCLICredential.
+type AzureDeveloperCLICredentialOptions struct {
+	// AdditionallyAllowedTenants specifies tenants for which the credential may acquire tokens, in addition
+	// to TenantID. Add the wildcard value "*" to allow the credential to acquire tokens for any tenant the
+	// logged in account can access.
+	AdditionallyAllowedTenants []string
+	// TenantID identifies the tenant the credential should authenticate in.
+	// Defaults to the azd environment, which is the tenant where selected Azure subscription is.
+	TenantID string
+
+	tokenProvider azureDeveloperCLITokenProvider
+}
+
+// init returns an instance of AzureDeveloperCLICredentialOptions initialized with default values.
+func (o *AzureDeveloperCLICredentialOptions) init() {
+	if o.tokenProvider == nil {
+		o.tokenProvider = defaultAzdTokenProvider
+	}
+}
+
+// AzureDeveloperCLICredential authenticates as the identity logged in to the Azure Developer CLI.
+type AzureDeveloperCLICredential struct {
+	s             *syncer
+	tokenProvider azureDeveloperCLITokenProvider
+}
+
+// NewAzureDeveloperCLICredential constructs an AzureDeveloperCLICredential. Pass nil to accept default options.
+func NewAzureDeveloperCLICredential(options *AzureDeveloperCLICredentialOptions) (*AzureDeveloperCLICredential, error) {
+	cp := AzureDeveloperCLICredentialOptions{}
+	if options != nil {
+		cp = *options
+	}
+	cp.init()
+	c := AzureDeveloperCLICredential{tokenProvider: cp.tokenProvider}
+	c.s = newSyncer(
+		credNameAzureDeveloperCLI,
+		cp.TenantID,
+		c.requestToken,
+		nil, // this credential doesn't have a silent auth method because the CLI handles caching
+		syncerOptions{AdditionallyAllowedTenants: cp.AdditionallyAllowedTenants},
+	)
+	return &c, nil
+}
+
+// GetToken requests a token from the Azure CLI. This credential doesn't cache tokens, so every call invokes the CLI.
+// This method is called automatically by Azure SDK clients.
+func (c *AzureDeveloperCLICredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (azcore.AccessToken, error) {
+	if len(opts.Scopes) != 1 {
+		return azcore.AccessToken{}, errors.New(credNameAzureDeveloperCLI + ": GetToken() requires exactly one scope")
```

Comment:
Is this true? The token provider handles multiple scopes

Reply:
you're right. tks.  I removed this